test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -0,0 +1,970 @@
+				imageRefs = append(imageRefs, b.imageRef)
+				bundleVersions = append(bundleVersions, b.bundleVersion)
+			}
+			fbc := CreateFBC(catalogDInfo.operatorName, catalogDInfo.desiredChannelName, imageRefs, bundleVersions)
+			err = WriteFBC(*fbc, catalogDInfo.baseFolderPath+"/"+catalogDInfo.catalogDir, catalogDInfo.fbcFileName)
+			Expect(err).To(Not(HaveOccurred()))
+		})
+	})
+	When("Build and load the FBC image into the test environment", func() {
+		It("Generate the docker file, build and load FBC image", func() {
+			By("Calling generate dockerfile function written")
+			err = generateDockerFile(catalogDInfo.baseFolderPath, catalogDInfo.catalogDir, catalogDInfo.catalogDir+".Dockerfile")
+			Expect(err).To(Not(HaveOccurred()))
+
+			By("Building the catalog image and loading into the test environment")
+			dockerContext := catalogDInfo.baseFolderPath
+			dockerfilePath := dockerContext + "/" + catalogDInfo.catalogDir + ".Dockerfile"
+			err = buildPushLoadContainer(catalogDInfo.imageRef, dockerfilePath, dockerContext, kindServer, GinkgoWriter)
+			Expect(err).To(Not(HaveOccurred()))
+		})
+	})
+	When("Create a catalog object and check if the resources are created", func() {
+		It("Create catalog object for the FBC and check if catalog, packages and bundle metadatas are created", func() {
+			bundleVersions := make([]string, len(bundleInfo.bundles))
+			for i, bundle := range bundleInfo.bundles {
+				bundleVersions[i] = bundle.bundleVersion
+			}
+			operatorCatalog, err = createCatalogCheckResources(operatorCatalog, catalogDInfo, bundleVersions)
+			Expect(err).To(Not(HaveOccurred()))
+		})
+	})
+	When("Install an operator and check if the operator operations succeed", func() {
+		It("Create an operator object and install it", func() {
+			By("Creating an operator object")
+			operator, err = createOperator(ctx, catalogDInfo.operatorName, operatorAction.installVersion)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking if the operator operations suceceded")
+			checkOperatorOperationsSuccess(operator, catalogDInfo.operatorName, operatorAction.installVersion, bundleInfo.baseFolderPath)
+		})
+	})
+	When("Upgrade an operator to higher version and check if the operator operations succeed", func() {
+		It("Upgrade to a higher version of the operator", func() {
+			By("Upgrading the operator")
+			operator, err = upgradeOperator(ctx, catalogDInfo.operatorName, operatorAction.upgradeVersion)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking if the operator operations succeeded")
+			checkOperatorOperationsSuccess(operator, catalogDInfo.operatorName, operatorAction.upgradeVersion, bundleInfo.baseFolderPath)
+		})
+	})
+	When("Delete an operator", func() {
+		It("Delete an operator", func() {
+			err = deleteOperator(ctx, catalogDInfo.operatorName)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verifying the operator doesn't exist")
+			Eventually(func(g Gomega) {
+				err = checkOperatorDeleted(catalogDInfo.operatorName)
```

Comment:
Given the func name `checkOperatorDeleted(...)`, it might make more sense for it to return a bool and do something like `return err != nil && errors.IsNotFound(err)` inside of the func, which would also simplify the checks here a bit.